pysparkling/tests/test_multiprocessing.py

```diff
@@ -128,18 +128,21 @@ def test_zipWithIndex(self):
         self.assertIn((4, 2), r)
 
     def test_cache(self):
-        r = self.sc.parallelize(range(3), 3)
+        to_check = list(range(5))
+        r = self.sc.parallelize(to_check, 3)
 
         def sleep05(v):
             time.sleep(0.5)
             return v
 
         r = r.map(sleep05).cache()
-        self.assertEqual(r.collect(), [0, 1, 2])
+        self.assertCountEqual(r.collect(), to_check)
```

Comment:
Could we modify the assertion to ensure that both contains the same items?

To ensure that for instance these int are not wrapped in an internal object by mistake

Reply:
assertCountEqual (sigh, got to explain this at work millions of times :p)... So counterintuitive name...

Basically it means that you compare both iterables to have the exact same elements, but not considering the order that they are in.

So [1, 2, 3] == [3, 2, 1].

It's my preference for spark code to test this way because of the parallelization inherent in it.

Reply:
RDDs should be guaranteed to be in order, no?

Reply:
As far as I know, no. In pysparkling maybe yes ;-), but in Spark you don't have too much control over how the framework determines what to run first on what executor... Unless you give a `sortBy()`, but that pulls everything to the driver.

If you just state `.collect()`, it will pull your information to the driver, first come, first serve.

At least that's what I always learned: never trust the order that you get information, unless you __explicitly__ asked for that order.